### PR TITLE
add `GetEpoch` local state query

### LINKED ISSUE #315: feat: implement `GetEpoch` local state query
DoD:

- [ ] we have primitives to represent the request
- [ ] we have primitives to represent the response
- [ ] we have golden tests using cbor test data

### PR DESCRIPTION
**Pull request by falcucci**
fixes #315

### CODE REVIEW

File: pallas-network/tests/protocols.rs [hunk 1]
```diff
@@ -399,6 +401,150 @@ pub async fn local_state_query_server_and_client_happy_path() {
     _ = tokio::join!(client, server);
 }
 
+#[tokio::test]
+pub async fn local_state_query_server_and_block_query_get_epoch_client_happy_path() {
```
Comment (scarmuega) [inline pallas-network/tests/protocols.rs:405]: there's no need to repeat the whole code of the test for each type of query. Put everything in a single happy path test that includes all of the available queries.

**Comment by scarmuega** [comment]
@falcucci we're missing part of the requirements of the issue: "we have primitives to represent the response". What I mean by that is that we need to get rid of the `GenericResponse` and start using structs that represent the decoded version of the query results.

File: pallas-network/tests/protocols.rs (hunk 1 above)
Comment (falcucci) [reply pallas-network/tests/protocols.rs]: @scarmuega done.

**Comment by scarmuega** [comment]
closing in favor of #326

**Pull request closed without merging**